src/abstractdataframe/where.jl

```diff
@@ -0,0 +1,181 @@
+using DataFrames: normalize_selection, ColumnIndex
+# where allows a transformation specification without a target column name or a column
+
+_process_where_pair(i::Int, a::Pair{<:Any, <:Base.Callable}) = first(a) => last(a) => Symbol(:x, i)
+_process_where_pair(i::Int, a::ColumnIndex) = a => Symbol(:x, i)
+_process_where_pair(i::Int, a) = throw(ArgumentError("Condition specifier $a is not supported by `where`"))
+
+function _get_where_conditions(df::Union{AbstractDataFrame, GroupedDataFrame},
+                               @nospecialize(args), dropmissing::Bool)
+    conditions = Any[_process_where_pair(i, a) for (i, a) in enumerate(args)]
+
+    isempty(conditions) && throw(ArgumentError("At least one condition must be passed"))
+
+    if df isa AbstractDataFrame
+        df_conditions = select(df, conditions..., copycols=!(df isa DataFrame))
+    else
+        df_conditions = select(df, conditions...,
+                               copycols=!(parent(df) isa DataFrame), keepkeys=false)
```

Comment:
I'm not sure I understand this. If `sdf isa SubDataFrame`, then `copycols` becomes `false`. But somehow this doesn't error.

Reply:
I think it is correct `parent(df) isa DataFrame` is `false` if the parent is a `SubDataFrame` and we negate it with `!` so:
* I copy columns for `SubDataFrame`
* do not copy columns for `DataFrame`